### PR TITLE
List of all lang items in unstable book.

### PR DESCRIPTION
**Pull request by Havvy**
I don't know how to link to the source code correctly, so I just put the information aside the lang item name.

### CODE REVIEW

**Comment by QuietMisdreavus** [comment]
cc @rust-lang/docs

I'm :+1: :+1: :+1: about having this list in here. I'm not sure about the presentation, though - it would be cool to have at least an intro statement saying that this is the list and where to find each definition. Adding a proper description to each item would be *ideal*, but possibly out of scope to just getting the list in there in the first place.

Also, is there some way to have tidy check for this? I don't think adding lang items is nearly as widespread as adding features (if it's even done any more), but if it were, it would be cool to make sure they were listed here. It would also help to somewhat automate putting the right source file in there, to make sure the docs don't diverge too much. I don't remember who was responsible for the current tidy unstable-book check - @steveklabnik was that you, or do you know?

**Comment by GuillaumeGomez** [comment]
It's a bit difficult to find what we're looking for. Maybe sort alphabetically?

**Comment by Havvy** [comment]
I've rebased with @QuietMisdreavus's requested change w.r.t. a sentence describing what the list is.

For @GuillaumeGomez's suggestion, I don't think fully alphabetical is useful. Instead, I put them in each sections. I guess I could have alphabetized each section though...I'll do that if I have to do some other change.

Really, this list is supposed to be a springboard for people actually documenting what each lang item is (and can easily be turned into a tracking issue) at some point in the future; probably after the impl period?

File: src/doc/unstable-book/src/language-features/lang-items.md [hunk 1]
```diff
@@ -227,3 +227,99 @@ A third function, `rust_eh_unwind_resume`, is also needed if the `custom_unwind_
...
+  - usize: libcore/num/mod.rs
+  - f32: libstd/f32.rs
+  - f64: libstd/f64.rs
+  - char: libstd_unicode/char.rs
+  - slice: liballoc/slice.rs
+  - str: liballoc/str.rs
+  - const_ptr: libcore/ptr.rs
+  - mut_ptr: libcore/ptr.rs
+  - unsafe_cell: libcore/cell.rs
+- Rutime
```
Comment (tshepang) [inline src/doc/unstable-book/src/language-features/lang-items.md]: should be `Runtime`

File: src/doc/unstable-book/src/language-features/lang-items.md [hunk 2]
```diff
@@ -99,7 +99,7 @@ libc = { version = "0.2.14", default-features = false }
 ```
 
 Note that the default features have been disabled. This is a critical step -
-**the default features of libc include the standard library and so must be
+- -*the default features of libc include the standard library and so must be
```
Comment (QuietMisdreavus) [inline src/doc/unstable-book/src/language-features/lang-items.md]: Looks like a misplaced change here?

File: src/doc/unstable-book/src/language-features/lang-items.md [hunk 3]
```diff
@@ -227,3 +227,99 @@ A third function, `rust_eh_unwind_resume`, is also needed if the `custom_unwind_
...
+  - mut_ptr: libcore/ptr.rs
+  - unsafe_cell: libcore/cell.rs
+- Runtime
+  - start: libstd/rt.rs
+  - eh_personality: libpanic_unwind/emcc.rs (EMCC)
+  - eh_personality: libpanic_unwind/seh64_gnu.rs (SEH64 GNU)
+  - eh_personality: libpanic_unwind/seh.rs (SEH)
+  - eh_unwind_resume: libpanic_unwind/seh64_gnu.rs (SEH64 GNU)
+  - eh_unwind_resume: libpanic_unwind/gcc.rs (GCC)
+  - msvc_try_filter": libpanic_unwind/seh.rs (SEH)
```
Comment (QuietMisdreavus) [inline src/doc/unstable-book/src/language-features/lang-items.md]: Errant quote?

**Review by QuietMisdreavus — COMMENTED** [review]
- Could you wrap the lang-item names and the paths in backticks, just so we don't have errant underscores mangling the output?
- I kinda feel like `coerce_unsized` belongs more with the "Other" items rather than with the rest of the operators, since it's not really describing an operator in the same way as `index` or `bitxor`, but a behavior. (It just happens to be defined alongside the rest of the operators. `>_>`)

File: src/doc/unstable-book/src/language-features/lang-items.md [hunk 4]
```diff
@@ -227,3 +227,99 @@ A third function, `rust_eh_unwind_resume`, is also needed if the `custom_unwind_
...
+  - `char`: `libstd_unicode/char.rs`
+  - `slice`: `liballoc/slice.rs`
+  - `str`: `liballoc/str.rs`
+  - `const_ptr`: `libcore/ptr.rs`
+  - `mut_ptr`: `libcore/ptr.rs`
+  - `unsafe_cell`: `libcore/cell.rs`
+- Runtime
+  - `start`: `libstd/rt.rs`
+  - `eh_personality`: `libpanic_unwind/emcc.rs (EMCC)`
+  - `eh_personality`: `libpanic_unwind/seh64_gnu.rs (SEH64 GNU)`
```
Comment (QuietMisdreavus) [inline src/doc/unstable-book/src/language-features/lang-items.md]: I guess these platform markers should go outside the backticks for the filename.

**Review by QuietMisdreavus — COMMENTED** [review]
Gets a :+1: from me with this, though I'd like to wait on @steveklabnik and @frewsxcv to make sure the whole docs team sees it before giving an r+.

File: src/doc/unstable-book/src/language-features/lang-items.md (hunk 4 above)
Comment (Havvy) [reply src/doc/unstable-book/src/language-features/lang-items.md]: Yes....

File: src/doc/unstable-book/src/language-features/lang-items.md [hunk 5]
```diff
@@ -227,3 +227,99 @@ A third function, `rust_eh_unwind_resume`, is also needed if the `custom_unwind_
...
+
+- Primitives
+  - `i8`: `libcore/num/mod.rs`
+  - `i16`: `libcore/num/mod.rs`
+  - `i32`: `libcore/num/mod.rs`
+  - `i64`: `libcore/num/mod.rs`
+  - `i128`: `libcore/num/mod.rs`
+  - `isize`: `libcore/num/mod.rs`
+  - `isize`: `libcore/num/mod.rs`
+  - `isize`: `libcore/num/mod.rs`
```
Comment (frewsxcv) [inline src/doc/unstable-book/src/language-features/lang-items.md]: looks like `isize` is repeated (same with `usize` below)

**Review by frewsxcv — APPROVED** [review]
minus my comments, looks great to me!

File: src/doc/unstable-book/src/language-features/lang-items.md (hunk 5 above)
Comment (Havvy) [reply src/doc/unstable-book/src/language-features/lang-items.md]: How did I not see that? (It's defined once for each configuration of pointer size.)

**Comment by frewsxcv** [comment]
thanks!

@bors r+ rollup

**Pull request merged**